src/comm/bench-flow.js

```diff
@@ -308,9 +313,17 @@ module.exports.run = async function(configFile, networkFile) {
             await execAsync(networkObject.caliper.command.start);
         }
 
+        if (configurationType) {
+            let blockListenerPath = path.join(__dirname, absCaliperPath, 'listener/block-listener-handler.js');
+            listener_child = childProcess.fork(blockListenerPath);
+            listener_child.on('error', function () {
+                logger.error('client encountered unexpected error');
+            });
+            listener_child.send({ type:'test', config: absNetworkFile });
+        }
         await blockchain.init();
         await blockchain.installSmartContract();
-        let numberOfClients = await client.init();
+        let numberOfClients = await client.init(demo, configFile, absCaliperDir, listener_child, networkFile);
```

Comment:
To handle the exit, we have to pass the params to the init function since the error cannot be return back to the bench-flow.js. Consume events function is asynchronous and to return from asynchronous call a callback function is used.